src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -475,4 +477,18 @@ public SourcePath getObjCGeneratedHeaderPath() {
   public SourcePath getOutputPath() {
     return ExplicitBuildTargetSourcePath.of(getBuildTarget(), outputPath);
   }
+
+  /**
+   * @return {@link SourcePath} to the .swiftmodule output from the compilation process. A
+   * swiftmodule file contains the public interface for a module, and is basically a binary file
+   * format equivalent to header files for a C framework or library.
+   *
+   * A swiftmodule file contains serialized ASTs (and possibly SIL), it conforms to
+   * Swift Binary Serialization Format, more details about this binary format can be found here:
+   * https://github.com/apple/swift/blob/7e6d62dae4bae4eb3737a6f76c0e51534c1bcca3/docs/Serialization.rst.
+   */
+  public SourcePath getSwiftModuleOutputPath() {
+    return ExplicitBuildTargetSourcePath.of(getBuildTarget(),
+      Paths.get(outputPath.toString(), moduleName + ".swiftmodule"));
```

Comment:
This can produce a different module name than what's actually produced by the compilation in the case of sanitisation. Instead, it should just return the precomputed `modulePath` like so:

    return ExplicitBuildTargetSourcePath.of(getBuildTarget(), modulePath);

Reply:
👍